### DISCUSSION TITLE
Support for ThreeJSScene layer

### DISCUSSION THREAD

**Discussion by harisbal** [visgl/deck.gl#8721] [Q&A]
Hello,
I noticed that in the Roadmap there was this idea for a ThreeJSScene layer.
Is this something that's still on the roadmap?

I am asking because of a use case where we need to [clip a 3D model](https://threejs.org/examples/webgl_clipping.html)
Could we somehow achieve this with the currently available layers?

Cheers

**Comment by ibgreen** [comment]
> a ThreeJSScene layer. Is this something that's still on the roadmap?

No

> we need to [clip a 3D model](https://threejs.org/examples/webgl_clipping.html) 
> Could we somehow achieve this with the currently available layers?

That type of clipping is done in the fragment shader, it is done by comparing (negative dot product) fragment position information in the fragment shader with a list of (projected) planes provided as uniforms, and discarding fragments if the fail the test.

It depends on your skill and the amount of time you have. You could certainly create a sub layer and modify the layers shaders, or perhaps this could even be packaged as an extension. 

https://github.com/mrdoob/three.js/blob/2ff77e4b335e31c108aac839a07401664998c730/src/renderers/shaders/ShaderChunk/clipping_planes_fragment.glsl.js#L19

**Comment by Claudio9701** [comment]
I was looking for something similar; this implementation was helpful for me: https://zhuanlan.zhihu.com/p/634390324
